src/Pos/Context/Functions.hs

```diff
@@ -43,8 +48,10 @@ import           Pos.Util.Context       (HasContext, askContext)
 genesisUtxoM :: (Functor m, WithNodeContext ssc m) => m Utxo
 genesisUtxoM = ncGenesisUtxo <$> getNodeContext
 
-genesisLeadersM :: (Functor m, WithNodeContext ssc m) => m SlotLeaders
-genesisLeadersM = ncGenesisLeaders <$> getNodeContext
+genesisLeadersM :: (MonadDB m, WithNodeContext ssc m) => m SlotLeaders
+genesisLeadersM = ifM isBootstrapEra
+    (pure $ NE.fromList $ HM.keys genesisBalances)
```

Comment:
@ChShersh the thing is that e. g. in dev mode `genesisBalances` may very simply be empty. It won't lead to a failure though, because the condition won't hold.